homeassistant/components/bayesian/helpers.py

```diff
@@ -0,0 +1,71 @@
+"""Helpers to deal with bayesian observations."""
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+import uuid
+
+from homeassistant.const import (
+    CONF_ABOVE,
+    CONF_BELOW,
+    CONF_ENTITY_ID,
+    CONF_PLATFORM,
+    CONF_VALUE_TEMPLATE,
+)
+from homeassistant.helpers.template import Template
+
+from .const import CONF_P_GIVEN_F, CONF_P_GIVEN_T, CONF_TO_STATE
+
+
+@dataclass
+class Observation:
+    """Representation of a sensor or template observation."""
+
+    entity_id: str | None
+    platform: str
+    prob_given_true: float
+    prob_given_false: float
+    to_state: str | None
+    above: float | None
+    below: float | None
+    value_template: Template | None
+    observed: bool | None = None
+    id: str = field(default_factory=lambda: str(uuid.uuid4()))
+
+    def to_dict(self) -> dict[str, str | float | bool | None]:
+        """Represent Class as a Dict for easier serialization. Sadly necessary because dataclasses asdict() can't serialize Templates and ignores Properties."""
+
+        dic = {
```

Comment:
irritating boilerplate but I'm keen to avoid adding attr as a dependency.
Defining a new `default_factory` doesn't work either as it is still applied recursively which breaks when it hits a template object

Reply:
I think you should split this into multiple lines.
Or maybe the "sadly necessary..." should be moved to a comment below instead of keeping inside the docstring.